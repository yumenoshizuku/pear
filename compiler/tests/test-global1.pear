def a,
def b,

Printa()
(
  return a,
)

Printb()
(
  return b,
)

Incab()
(
  a = a + 1,
  b = b + 1,
  return a + b,
)

Main()
(
  a = 42,
  b = 21,

  printa = Printa(),
  printb = Printb(),
  incab = Incab(),

  puts(printa()),
  puts(printb()),
  puts(incab()),
  puts(printa()),
  puts(printb()),
)
