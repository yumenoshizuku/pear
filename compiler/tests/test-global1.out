#include <stdio.h>
#include <stdlib.h>
#include <gtk/gtk.h>
#include <glib.h>
#include <glib/gstdio.h>


int main(int argc, char** argv)
{
	
	gtk_init(&argc, &argv);
	printf("%d\n", 42);
	printf("%d\n", 21);
	printf("%d\n", 65);
	printf("%d\n", 43);
	printf("%d\n", 22);
	gtk_main();
	return 0;
}

